File: LoadGenerator/Album.py
"""
░█████╗░██╗░░░░░██████╗░██╗░░░██╗███╗░░░███╗
██╔══██╗██║░░░░░██╔══██╗██║░░░██║████╗░████║
███████║██║░░░░░██████╦╝██║░░░██║██╔████╔██║
██╔══██║██║░░░░░██╔══██╗██║░░░██║██║╚██╔╝██║
██║░░██║███████╗██████╦╝╚██████╔╝██║░╚═╝░██║
╚═╝░░╚═╝╚══════╝╚═════╝░░╚═════╝░╚═╝░░░░░╚═╝
"""

class Album:

    def __init__(self, id, name, release_date, album_type, totaltrack, uri,
                 artists):
        self.id = id
        self.name = name
        self.release_date = release_date
        self.totalTrack = totaltrack
        self.album_type = album_type
        self.uri = uri
        self.artists = artists

    def to_dict(self):
        artists = [artist.to_dict() for artist in self.artists]
        return {"id": self.id, "name": self.name, "release_date": self.release_date, "totalTrack": self.totalTrack,
                "album_type": self.album_type, "uri": self.uri, "artists": artists
                }

File: MSSpotiAPI/src/main/java/com/ms/spotiapi/Models/Album.java
package com.ms.spotiapi.Models;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.io.Serializable;
import java.util.HashSet;
import java.util.Set;

@Entity
@Table(name="ALBUMS")
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class Album implements Serializable {
    @Id
    private String id;

    private String name;

    private String release_date;

    private int total_track;

    private String album_type;

    private String uri;

    @ManyToMany
    @JoinTable(
            name = "album_artists",
            joinColumns = @JoinColumn(name="album_name",referencedColumnName = "name"),
            inverseJoinColumns = @JoinColumn(name="artist_name",referencedColumnName = "name")
    )
    public Set<Artist> artists = new HashSet<>();

}
